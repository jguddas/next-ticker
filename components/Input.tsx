import React from 'react'
import { x } from '@xstyled/styled-components'

import Text from './Text'

interface Props extends React.ComponentProps<typeof x.div> {
  title: string
}

export default ({ title, ...props }: Props): JSX.Element => (
  <x.div display="flex" flexDirection="column" flex="1" {...props}>
    <Text mb={3}>{title}</Text>
    <x.input
      w="100%"
      minWidth="0px"
      flex="1"
      px={5}
      py={4}
      fontSize="input"
      lineHeight
      outline="none"
      color="input"
      bg="inputBackground"
    />
  </x.div>
)
