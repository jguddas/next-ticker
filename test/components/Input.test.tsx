import React from 'react'
import { render } from '../testUtils'
import Input from '../../components/Input'
import 'jest-styled-components'

describe('Input', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Input title="Hello World" />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
