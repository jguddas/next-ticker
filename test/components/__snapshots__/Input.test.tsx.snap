// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<DocumentFragment>
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1.c1 {
  color: text;
  font-size: 1px;
  margin-bottom: 3px;
}

.c2.c2 {
  -webkit-transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  width: 100%;
  min-width: 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 5px;
  padding-left: 5px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: input;
  outline: none;
  color: input;
  background-color: inputBackground;
}

.c2.c2:focus {
  --x-shadow: inputFocus;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      Hello World
    </div>
    <input
      class="c2"
    />
  </div>
</DocumentFragment>
`;
